sqlglot/generator.py

```diff
@@ -677,11 +677,12 @@ def qualify_sql(self, expression):
     def union_sql(self, expression):
         return self.prepend_ctes(
             expression,
-            self.set_operation(
-                expression, f"UNION{'' if expression.args.get('distinct') else ' ALL'}"
-            ),
+            self.set_operation(expression, self.union_op(expression)),
         )
 
+    def union_op(self, expression):
+        return f"UNION{'' if expression.args.get('distinct') else ' ALL'}"
```

Comment:
should we just explicitly make union distinct? then we don't need the bigquery edge case?

Reply:
I don't think this is safe in general, e.g. [Snowflake](https://docs.snowflake.com/en/sql-reference/operators-query.html) doesn't accept the DISTINCT syntax. This is the grammar for Snowflake:
```
{ INTERSECT | { MINUS | EXCEPT } | UNION [ ALL ] }
```

Reply:
[Postgres](https://www.postgresql.org/docs/current/queries-union.html) is the same:
```
query1 UNION [ALL] query2
query1 INTERSECT [ALL] query2
query1 EXCEPT [ALL] query2
```

Reply:
great, thanks for the research